components/content/src/utils.rs

```diff
@@ -149,6 +197,120 @@ mod tests {
+            "/posts/my-article/subdir/GGG.txt",
+        ];
+
+        let serialized_assets =
+            serialize_assets(&assets, Some(&page_folder_path), Some(&colocated_path));
+
+        assert_eq!(
+            serialized_assets, expected_serialized_assets,
+            "Serialized assets (left) are different from expected (right)",
+        );
+    }
+
+    #[test]
+    fn can_serialize_empty_assets() {
+        let parent_path = Path::new("/tmp/test");
+        let page_folder_path = parent_path.join("content").join("posts").join("my-article");
+        let assets: Vec<PathBuf> = vec![];
+
+        let serialized_assets = serialize_assets(&assets, Some(&page_folder_path), None);
+
+        assert!(serialized_assets.is_empty());
+    }
+
+    #[test]
+    fn can_get_assets_permalinks() {
+        let serialized_assets = vec![
+            "/posts/my-article/example.js".to_string(),
+            "/posts/my-article/graph.jpg".to_string(),
+            "/posts/my-article/fail.png".to_string(),
+            "/posts/my-article/extensionless".to_string(),
+            "/posts/my-article/subdir/example.js".to_string(),
+            "/posts/my-article/FFF.txt".to_string(),
+            "/posts/my-article/GRAPH.txt".to_string(),
+            "/posts/my-article/subdir/GGG.txt".to_string(),
+        ];
+        let parent_permalink = "https://remplace-par-ton-url.fr/posts/my-super-article/";
+        let colocated_path = "posts/my-article/".to_string();
+        let mut expected_assets_permalinks = HashMap::<String, String>::new();
+        expected_assets_permalinks.insert(
+            "posts/my-article/example.js".to_string(),
+            format!("{}{}", parent_permalink, "example.js"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/graph.jpg".to_string(),
+            format!("{}{}", parent_permalink, "graph.jpg"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/fail.png".to_string(),
+            format!("{}{}", parent_permalink, "fail.png"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/extensionless".to_string(),
+            format!("{}{}", parent_permalink, "extensionless"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/subdir/example.js".to_string(),
+            format!("{}{}", parent_permalink, "subdir/example.js"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/FFF.txt".to_string(),
+            format!("{}{}", parent_permalink, "FFF.txt"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/GRAPH.txt".to_string(),
+            format!("{}{}", parent_permalink, "GRAPH.txt"),
+        );
+        expected_assets_permalinks.insert(
+            "posts/my-article/subdir/GGG.txt".to_string(),
+            format!("{}{}", parent_permalink, "subdir/GGG.txt"),
+        );
```

Comment:
Something like:

```
let m: HashMap<_, _> = HashMap::from_iter([("one", 1), ("two", 2), ("three", 3)]);
```

is cleaner

Reply:
Thanks for the head's up.
I'll fix it shortly.

Reply:
OK, I had time to fix it right away in the end.